DETAILED_INSTALL.md

```diff
@@ -0,0 +1,295 @@
+    ```
+
+    For example, `make run example=examples/petting_zoo/sisl/multiwalker/feedforward/decentralised/run_mad4pg.py`.
+
+    Alternatively, run bash inside a docker container with mava installed, `make bash`, and from there examples can be run as follows: `python dir/to/example/example.py`.
+
+    To run an example with tensorboard viewing enabled, you can run
+
+    ```bash
+    make run-tensorboard example=dir/to/example/example.py
+    ```
+
+    and navigate to `http://127.0.0.1:6006/`.
+
+    To run an example where agents are recorded (**ensure you built the image with `record=true`**):
+
+    ```
+    make run-record example=dir/to/example/example.py
+    ```
+
+    Where example, is an example with recording available e.g. `examples/debugging/simple_spread/feedforward/decentralised/run_maddpg_record.py`.
+
+### Python virtual environment
+
+1. If not using docker, we strongly recommend using a
+    [Python virtual environment](https://docs.python.org/3/tutorial/venv.html)
+    to manage your dependencies in order to avoid version conflicts. Please note that since Launchpad only supports Linux based OSes, using a python virtual environment will only work in these cases:
+
+    ```bash
+    python3 -m venv mava
+    source mava/bin/activate
+    pip install --upgrade pip setuptools
+    ```
+
+    1.1  To install the core libraries, including [Reverb](https://github.com/deepmind/reverb) - our storage dataset , Tensorflow and [Launchpad](https://github.com/deepmind/launchpad) - for distributed agent support :
+
+    * Install swig for box2d:
+
+    ```bash
+    sudo apt-get install swig -y
+    ```
+
+    * Install core dependencies:
+
+    ```bash
+    pip install id-mava[tf,reverb]
+    ```
+
+    * Or for the latest version of mava from source (**you can do this for all pip install commands below for the latest depedencies**):
+
+    ```bash
+    pip install git+https://github.com/instadeepai/Mava#egg=id-mava[reverb,tf]
+    ```
+
+    **For the jax version of mava, please replace `tf` with `jax`, e.g. `pip install id-mava[jax,reverb]`**.
+
+    **If you are using `jax` and `CUDA`, you also need to run the following:**
+
+    ```bash
+    pip install --upgrade "jax[cuda]" -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html
+    ```
+
+    1.2 For **optional** environments:
+    * PettingZoo:
+
+        ```
+        pip install id-mava[pz]
+        ```
+
+    * Flatland:
+
+        ```
+        pip install id-mava[flatland]
+        ```
+
+    * Openspiel:
+
+        ```
+        pip install id-mava[open_spiel]
+        ```
+
+    * 2D RoboCup environment:
+
+        A local install has only been tested using the Ubuntu 18.04 operating system.
+        The installation can be performed by running the RoboCup bash script while inside the Mava
+        python virtual environment.
+
+        ```bash
+        ./bash_scripts/install_robocup.sh
+        ```
+
+    * StarCraft II:
```

Comment:
Thanks @arnupretorius for the PR!
Missing an instruction `pip install .[sc2]`

Reply:
Thanks @OmaymaMahjoub :) I added the instruction.